codec-redis/src/main/java/io/netty/handler/codec/redis/BulkStringRedisMessage.java

```diff
@@ -0,0 +1,170 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.DefaultByteBufHolder;
+import io.netty.buffer.Unpooled;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Bulk Strings of <a href="http://redis.io/topics/protocol">RESP</a>.
+ */
+public class BulkStringRedisMessage extends DefaultByteBufHolder implements RedisMessage {
+
+    BulkStringRedisMessage(ByteBuf content) {
+        super(content);
+    }
+
+    @Override
+    public RedisMessageType type() {
+        return RedisMessageType.BULK_STRING;
+    }
+
+    @Override
+    public boolean isNull() {
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return new StringBuilder(StringUtil.simpleClassName(this))
+                .append('[')
+                .append("content=")
+                .append(content())
+                .append(']').toString();
+    }
+
+    static final BulkStringRedisMessage NULL_INSTANCE = new BulkStringRedisMessage(Unpooled.EMPTY_BUFFER) {
+        @Override
+        public boolean isNull() {
+            return true;
+        }
+
+        @Override
+        public ByteBuf content() {
+            return null;
+        }
+
+        @Override
+        public BulkStringRedisMessage copy() {
+            return NULL_INSTANCE;
+        }
+
+        @Override
+        public BulkStringRedisMessage duplicate() {
+            return this;
+        }
+
+        @Override
+        public int refCnt() {
+            return 1;
+        }
+
+        @Override
+        public BulkStringRedisMessage retain() {
+            return this;
+        }
+
+        @Override
+        public BulkStringRedisMessage retain(int increment) {
+            return this;
+        }
+
+        @Override
+        public BulkStringRedisMessage touch() {
+            return this;
+        }
+
+        @Override
+        public BulkStringRedisMessage touch(Object hint) {
+            return this;
+        }
+
+        @Override
+        public boolean release() {
+            return false;
+        }
+
+        @Override
+        public boolean release(int decrement) {
+            return false;
+        }
+
+        @Override
+        public String toString() {
+            return "NullBulkStringRedisMessage";
+        }
+    };
+
+    static final BulkStringRedisMessage EMPTY_INSTANCE = new BulkStringRedisMessage(Unpooled.EMPTY_BUFFER) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 56 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1188)